Add patient class and features

xhphoong opened the pull request:
Minor changes left

Commit:
Add gender attribute to person

Commit:
Add patient class

HAVE NOT COMPLETED

Let's:
- Add patient class
- Add date and time attributes to patient
- The modification needed for other classes is only partially done

LolfoollorS commented on src/main/java/seedu/address/model/person/Patient.java:
```diff
@@ -0,0 +1,75 @@
+        } else {
+            return super.isSamePerson(otherPerson);
+        }
+    }
```
Maybe not needed since a person is still the same person if he/she is a patient or not

LolfoollorS commented on src/main/java/seedu/address/model/person/Patient.java:
```diff
@@ -0,0 +1,75 @@
+        }
+    }
+
+    /**
+     * Returns true if both patients have the same identity and data fields.
+     * This defines a stronger notion of equality between two persons.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (!(other instanceof Patient)) {
+            return false;
+        } else {
+            Patient otherPatient = (Patient) other;
+            return super.equals(other) && otherPatient.getDate().equals(getDate())
+                    && otherPatient.getTime().equals(getTime());
+        }
+    }
```
Similar to before, maybe this is not needed as there is already a isSamePerson()

LolfoollorS commented on src/main/java/seedu/address/model/person/Patient.java:
```diff
@@ -0,0 +1,75 @@
+        }
+    }
+
+    @Override
+    public int hashCode() {
+        // use this method for custom fields hashing instead of implementing your own
+        return super.hashCode() + Objects.hash(date, time);
+    }
```
Similar to before, there is already a hashCode()

LolfoollorS reviewed (commented):
LGTM

Commit:
Add patient class and modify accordingly

Let's,
-Add a patient class that extends from person class
-Add an extra attributes to the patient class: DateTime of Home-Visit.
-It can take in 0,1 or multiple DateTime of Home-Visit and display the
 sorted DateTime.
-Modify the parser and commands classes accordingly.
-Modify the storage classes accordingly.
-Modify the testcases accordingly.
-Fix the checkstyle.

Commit:
Fix the test error and checkstyle error

xhphoong committed:
Merge branch 'master' into add-patient-class-and-features

xhphoong commented:
ERROR

Commit:
Merge branch 'add-patient-class-and-features' of https://github.com/xhphoong/tp into add-patient-class-and-features

Commit:
Fix error after merging

xhphoong committed:
Merge branch 'master' into add-patient-class-and-features

Commit:
Fix error after merging

Commit:
Simplify the get indicator part

Commit:
Modify the add feature : the uid for new person is created internally

The user do not need to give the uid when adding each person. Now,
the uid will be created by the program itself and assign to the person
to be added.
